Source/WebCore/crypto/mac/CryptoKeyECMac.cpp

```diff
@@ -252,16 +251,27 @@ RefPtr<CryptoKeyEC> CryptoKeyEC::platformImportSpki(CryptoAlgorithmIdentifier id
     index += oidSize + 1; // Read named curve OID, BIT STRING
     if (keyData.size() < index + 1)
         return nullptr;
-    index += bytesUsedToEncodedLength(keyData[index]) + 1; // Read length, InitialOctet
-
-    if (!doesUncompressedPointMatchNamedCurve(curve, keyData.size() - index))
-        return nullptr;
-
+    index += bytesUsedToEncodedLength(keyData[index]); // Read length
+    uint8_t initialOctet = keyData[index];
+    index += 1;
+        
+        
+    bool uncompressed = initialOctet == 4 or initialOctet == 04 ;
+    bool compressed = initialOctet == 03 or initialOctet == 02;
+        
+        
     CCECCryptorRef ccPublicKey = nullptr;
-    if (CCECCryptorImportKey(kCCImportKeyBinary, keyData.data() + index, keyData.size() - index, ccECKeyPublic, &ccPublicKey))
-        return nullptr;
-
-    return create(identifier, curve, CryptoKeyType::Public, PlatformECKeyContainer(ccPublicKey), extractable, usages);
+        
+    if ( uncompressed ){
+    if ( CCECCryptorImportKey(kCCImportKeyBinary, keyData.data() + index, keyData.size() - index, ccECKeyPublic, &ccPublicKey))
+          return nullptr;
+    }
+        
+    if( compressed ){
+    if (CCECCryptorImportKey(kCCImportKeyCompact, keyData.data() + index, keyData.size() - index, ccECKeyPublic, &ccPublicKey))
+          return nullptr;
+      }
```

Comment:
With the way it is currently written, if initialOctet is 1, we would probably use a nullptr public key.
In the past code path, we would try to use kCCImportKeyBinary.
Either we need to return early if octet is not 2 3 or 4.
Or we could write it in the following way:
`if (compressed) {
      use kCCImportKeyCompact
} else {
     use existing code path (kCCImportKeyBinary)`
}

Reply:
Already changed to check if it is compressed or not and then use CCECCryptorImportKey with either  kCCImportKeyBinary or kCCImportKeyCompact.